### PR TITLE
Support AWQ & GroupWiseQuant for LLMs

### PR DESCRIPTION
**Pull request by RachelXu7**
### PR types

New features
### PR changes

Others
### Description

Support AWQ & GroupWiseQuant for LLMs

### CODE REVIEW

File: llm/quant.py [hunk 1]
```diff
@@ -123,6 +133,12 @@ def apply_smooth(quant_args, trainer, ptq_dataloader, ptq_model_config):
             weight_quant_method="abs_max_channel_wise",
             act_quant_method="avg",
         )
+    elif quant_args.do_awq:
+        search_func = AWQSearch(
+            n_grid=20,
+            bits_length=4,
```
Comment (wawltor) [inline llm/quant.py:139]: 现在AWQ只能支持4bit吗？bits_length=4是一个magic number，比较疑惑

File: llm/quant.py [hunk 2]
```diff
@@ -23,6 +23,8 @@
 from paddle.quantization import PTQ, QAT, QuantConfig
 from paddleslim.quant.advanced import (
     GPTQ,
+    AutoClip,
```
Comment (lugimzzz) [inline llm/quant.py:26]: 这里autoclip和awqsearch指的是什么算法？这块同步更新一下文档中算法介绍和参数配置吧https://github.com/PaddlePaddle/PaddleNLP/blob/develop/llm/docs/quantization.md

**Review by lugimzzz — DISMISSED** [review]
lgtm

**Review by lugimzzz — APPROVED** [review]
lgtm

**Pull request merged**